### PR TITLE
CMS tutorial in french, Installation, Configuration, Migration & update guides

### PR DESCRIPTION
**Pull request by o-dlr-o**
I am using an 'opinionated' translation for 'Policy'

### CODE REVIEW

File: en/tutorials-and-examples/cms/database.rst [hunk 1]
```diff
@@ -149,7 +149,7 @@ might look something like the following::
         // More configuration below.
     ];
 
-Once you've saved your **config/app.php** file, you should see that 'CakePHP is
+Once you've saved your **config/app.php** file, you should see that CakePHP is
 able to connect to the database' section have a green chef hat.
```
Comment (garas) [inline en/tutorials-and-examples/cms/database.rst]: The quoted string is  _‘CakePHP is able to connect to the database’_ so quote mark is not mistake.

File: fr/tutorials-and-examples/cms/authentication.rst [hunk 2]
```diff
@@ -2,181 +2,297 @@ Tutoriel CMS - Authentification
...
+
+Si vous vous en souvenez, par le passé nous utilisions :doc:`AuthComponent </controllers/components/authentication>`
+afin de gérer toutes ces étapes. A présent, la logique est divisée en classes spécifiques et
+le processus d'authentification se déroule avant votre couche de contrôleur. Il vérifie d'abord si l'utilisateur
+est authentifié (en fonction de la configuration que vous avez fournie) et injecte l'utilisateur ainsi que le
+résultat de l'authentification dans la requête afin que vous puissiez les utiliser ultérieurement.
+
+Dans **src/Application.php**, ajoutez les imports suivants::
+
+ // In src/Application.php add the following imports
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/authentication.rst:114]:
```suggestion
    // In src/Application.php add the following imports
```
The comment needs to be whitespace aligned with the text for sphinx to format correctly.

File: fr/tutorials-and-examples/cms/authentication.rst [hunk 3]
```diff
@@ -2,181 +2,297 @@ Tutoriel CMS - Authentification
...
+    use Authentication\AuthenticationService;
+    use Authentication\AuthenticationServiceInterface;
+    use Authentication\AuthenticationServiceProviderInterface;
+    use Authentication\Middleware\AuthenticationMiddleware;
+    use Psr\Http\Message\ServerRequestInterface;
+
+Ensuite, implémentez l'interface d'authentification pour votre classe ``Application```::
+
+ // dans src/Application.php
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/authentication.rst:123]:
```suggestion
    // dans src/Application.php
```
The comment needs to be whitespace aligned with the text for sphinx to format correctly.

File: fr/tutorials-and-examples/cms/authentication.rst [hunk 4]
```diff
@@ -2,181 +2,297 @@ Tutoriel CMS - Authentification
...
+Le résultat de l'authentification sera injecté dans un attribut de requête nommé
+``authentification``. Vous pouvez consulter le résultat à tout moment en utilisant
+``$this->request->getAttribute('authentication')`` à partir des actions de votre contrôleur.
+Toutes vos pages seront restreintes car le ``AuthenticationComponent`` vérifie le
+résultat à chaque demande. Lorsqu'il ne parvient pas à trouver un utilisateur authentifié, il redirige
+l'utilisateur sur la page ``/users/login``.
+Notez qu'à ce stade, le site ne fonctionnera pas car nous n'avons pas encore de page de connexion.
+Si vous visitez votre site, vous obtiendrez une "boucle de redirection infinie", allons corriger cela.
+
+Dans votre ``UsersController`, ajoutez le code suivant::
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/authentication.rst]:
```suggestion
Dans votre ``UsersController``, ajoutez le code suivant::
```

File: fr/tutorials-and-examples/cms/authentication.rst [hunk 5]
```diff
@@ -185,178 +301,24 @@ Si vous n'êtes pas connecté et essayez de visiter **/users/add**, vous serez
...
-de vous diriger vers la section :doc:`topics </topics>`.
+    // Ajoutez la méthode beforeFilter au UsersController
+    $this->Authentication->addUnauthenticatedActions(['login', 'add']);
+
+The above tells ``AuthenticationComponent`` that the ``add()`` action of the
+``UsersController`` does *not* require authentication or authorization. You may
+want to take the time to clean up the **Users/add.php** and remove the
+misleading links, or continue on to the next section. We won't be building out
+user editing, viewing or listing in this tutorial, but that is an exercise you
+can complete on your own.
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/authentication.rst]: Should this still be here?

File: fr/tutorials-and-examples/cms/database.rst [hunk 6]
```diff
@@ -52,7 +55,57 @@ pour créer les premières tables nécessaires au tutoriel::
 
     INSERT INTO articles (user_id, title, slug, body, published, created, modified)
     VALUES
-    (1, 'First Post', 'first-post', 'This is the first post.', 1, now(), now());
+    (1, 'First Post', 'first-post', 'This is the first post.', 1, NOW(), NOW());
+
+Si vous utilisez PostgreSQL, connectez-vous à la base de données ``cake_cms` et exécutez le
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/database.rst]:
```suggestion
Si vous utilisez PostgreSQL, connectez-vous à la base de données ``cake_cms`` et exécutez le
```

File: fr/tutorials-and-examples/cms/tags-and-users.rst [hunk 7]
```diff
@@ -388,14 +378,25 @@ suivante::
 
     echo $this->Form->control('tag_string', ['type' => 'text']);
 
+
+Nous devrons également mettre à jour le modèle de vue d'article. Dans
+**templates/Articles/view.php**, ajoutez la ligne comme indiqué:
```
Comment (markstory) [inline fr/tutorials-and-examples/cms/tags-and-users.rst]:
```suggestion
**templates/Articles/view.php**, ajoutez la ligne comme indiqué::
```

**Review by markstory — COMMENTED** [review]
Thank you, this looks great. I left a few formatting related updates that need to be applied.

File: fr/tutorials-and-examples/cms/authentication.rst (hunk 5 above)
Comment (o-dlr-o) [reply fr/tutorials-and-examples/cms/authentication.rst]: No i forgot it sorry it was a long day...

File: fr/development/routing.rst [hunk 8]
```diff
@@ -856,11 +889,26 @@ n'importe quelle sous-domaine::
...
             ['controller' => 'Images', 'action' => 'oldLogo']
-        )->setHost('images.example.com');
+        )->setHost('*.example.com');
     });
 
-L'option ``_host`` n'affecte que le parsing des URL depuis les requêtes et
-n'intervient jamais dans la génération d'URL.
+L'option ``_host`` est également utilisée dans la génération d'URL. Si votre option ``_host``
+spécifie un domaine exact, ce domaine sera inclus dans l'URL générée.
+Cependant, si vous utilisez un caractère générique, vous devrez fournir le ``_host`
```
Comment (markstory) [inline fr/development/routing.rst]:
```suggestion
Cependant, si vous utilisez un caractère générique, vous devrez fournir le ``_host``
```

File: fr/development/routing.rst [hunk 9]
```diff
@@ -905,13 +944,13 @@ toutes les routes connectées avant cet appel n'hériteront pas de ces extension
 
 En utilisant des extensions, vous dites au router de supprimer toutes les
 extensions de fichiers correspondant, puis d'analyser le reste. Si vous
-souhaitez créer une URL comme ``/page/title-of-page.html`` vous devriez créer
+souhaitez créer une URL comme /page/title-of-page.html vous devriez créer
```
Comment (markstory) [inline fr/development/routing.rst]:
```suggestion
souhaitez créer une URL comme ``/page/title-of-page.html`` vous devriez créer
```

File: fr/development/routing.rst [hunk 10]
```diff
@@ -1274,18 +1317,17 @@ tableau::
 
     ['controller' => 'Articles', 'action' => 'view', 5]
 
-Comme ``5`` a une clé numérique, il est traité comme un argument passé.
+Comme ``5`` poss!de une clé numérique, il est traité comme un argument passé.
```
Comment (markstory) [inline fr/development/routing.rst:1320]: Should this `!` be here?

File: fr/development/routing.rst [hunk 11]
```diff
@@ -1335,20 +1410,144 @@ vous générez des URLs:
...
+
+À titre d'exemple, imaginez un blog permettant aux utilisateurs de créer des **Articles** et
+**Commentaires**, et de marquer les deux comme *publié* ou *brouillon*. Les deuxpages d'index
+pourrait inclure l'ID utilisateur. L'URL **Commentaires** pourrait également inclure
+un identifiant d'article pour identifier l'article auquel le commentaire fait référence.
+
+Voici les URL correspondant à ce scénario::
+
+   /articles/index/42
+    /comments/index/42/18
```
Comment (markstory) [inline fr/development/routing.rst]:
```suggestion
  /comments/index/42/18
```

**Comment by o-dlr-o** [comment]
Thank you for your patience and your attention
regards

**Comment by markstory** [comment]
Thank you for the great updates :bow:

**Pull request merged**